tools/flaky_tests/Jenkinsfile

```diff
@@ -0,0 +1,63 @@
+// -*- mode: groovy -*-
+
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+err = null
+
+node('mxnetlinux-cpu') {
+  // Loading the utilities requires a node context unfortunately
+  checkout scm
+  utils = load('ci/Jenkinsfile_utils.groovy')
+}
+utils.assign_node_labels(linux_gpu: 'mxnetlinux-gpu')
+
+
+utils.main_wrapper(
+core_logic: {
+    stage('Preprocessing'){
+        node(NODE_LINUX_GPU){
```

Comment:
Please revise the node types. Also, make sure that the following stages are only executed if actually necessary.

Reply:
Please also note that the file tests.tmp will not persist in between nodes.